test/python/test_visualization.py

```diff
@@ -43,75 +39,127 @@
-        qc.h(qr[0])
-        qc.cx(qr[0], qr[1])
-        qc.measure(qr[1], cr[1])
-        qc.x(qr[1]).c_if(cr, 1)
-        qc.measure(qr, cr)
-        self.qp = qp
-        self.qc = qc
-        self.qobj = qp.compile(['latex_test'])
-
-    def test_latex_drawer(self):
-        filename = self._get_resource_path('test_latex_drawer.tex')
+    def randomCircuit(self, width, depth, max_operands):
+        """Generate random circuit of arbitrary size.
+        Note: the depth is the layers of independent operation. true depth
+        in the image may be more for visualization purposes, if gates overlap.
+
+        Args:
+            width (int): number of quantum wires
+            depth (int): layers of operations
+            max_operands (int): maximum operands of each gate
+
+        Returns:
+            QuantumCircuit: constructed circuit
+        """
+        width = 3
+        depth = 3
+        max_operands = 3
+
+        qr = qiskit.QuantumRegister("q", width)
+        qc = qiskit.QuantumCircuit(qr)
+
+        one_q_ops = "iden,u0,u1,u2,u3,x,y,z,h,s,sdg,t,tdg,rx,ry,rz"
+        two_q_ops = "cx,cy,cz,ch,crz,cu1,cu3,swap"
+        three_q_ops = "ccx"
+
+        # apply arbitrary random operations at every depth
+        for _ in range(depth):
+            # choose either 1, 2, or 3 qubits for the operation
+            remaining_qubits = list(range(width))
+            while remaining_qubits:
+                max_possible_operands = min(len(remaining_qubits), max_operands)
+                num_operands = random.choice(range(max_possible_operands))+1
+                operands = random.sample(remaining_qubits, num_operands)
+                remaining_qubits = [q for q in remaining_qubits if q not in operands]
+                if num_operands == 1:
+                    op = random.choice(one_q_ops.split(','))
+                elif num_operands == 2:
+                    op = random.choice(two_q_ops.split(','))
+                elif num_operands == 3:
+                    op = random.choice(three_q_ops.split(','))
+                # every gate is defined as a method of the QuantumCircuit class
+                # the code below is so we can call a gate by its name
+                gate = getattr(qiskit.QuantumCircuit, op)
+                op_args = list(signature(gate).parameters.keys())
+                num_angles = len(op_args) - num_operands - 1    # -1 for the 'self' arg
+                angles = [random.uniform(0, 3.14) for x in range(num_angles)]
+                register_operands = [qr[i] for i in operands]
+                gate(qc, *angles, *register_operands)
+
+        return qc
+
+    def test_tiny_circuit(self):
+        filename = self._get_resource_path('test_tiny.tex')
+        qc = self.randomCircuit(1, 1, 1)
+        try:
+            latex_drawer(qc, filename)
+        except Exception:
+            if os.path.exists(filename):
+                os.remove(filename)
+            raise
```

Comment:
Is the fact that the files are not removed if there is no exception raised intentional?

It's not a big issue as we are already cluttering the `test/python` folder a bit in a couple of other tests (but we can tackle that in another PR) - if it was not intentional, can you change them to something like:
```
        try:
            latex_drawer(qc, filename)
        finally:
            if os.path.exists(filename):
                os.remove(filename)
```

Reply:
it was not intentional, thanks for catching it.